gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -1248,4 +1249,56 @@ protected List<Statement> createTypeRegistry(Service service) {
                 .setValueExpr(typeRegistryBuilderExpr)
                 .build()));
   }
+
+  @Override
+  protected List<MethodDefinition> createClassMethods(
+      GapicContext context,
+      Service service,
+      TypeStore typeStore,
+      Map<String, VariableExpr> classMemberVarExprs,
+      Map<String, VariableExpr> callableClassMemberVarExprs,
+      Map<String, VariableExpr> protoMethodNameToDescriptorVarExprs,
+      List<Statement> classStatements) {
+    List<MethodDefinition> javaMethods = new ArrayList<>();
+    javaMethods.addAll(
+        super.createClassMethods(
+            context,
+            service,
+            typeStore,
+            classMemberVarExprs,
+            callableClassMemberVarExprs,
+            protoMethodNameToDescriptorVarExprs,
+            classStatements));
+    javaMethods.addAll(createInvalidClassMethods(service));
+    return javaMethods;
+  }
+
+  private List<MethodDefinition> createInvalidClassMethods(Service service) {
+    List<MethodDefinition> methodDefinitions = new ArrayList<>();
+    for (Method protoMethod : service.methods()) {
+      if (!protoMethod.isSupportedByTransport(getTransport())) {
```

Comment:
Usually for scenarios like this we can do something like 
```
if (protoMethod.isSupportedByTransport(getTransport())) {
  continue;
}
//do other logic...
```
to have one less indent so that the code is more readable.

Reply:
Gotcha, will update.